src/seeds/types.ts

```diff
@@ -0,0 +1,107 @@
+import { MusicPlatformType } from '../types/platform';
+
+enum SeedEntities {
+  'platforms',
+  'nftFactories',
+  'artists',
+  'processedTracks'
+}
+type SeedEntity = keyof typeof SeedEntities;
+
+const PlatformRequiredKeys = ['id', 'name', 'type'];
+const PlatformValidKeys = PlatformRequiredKeys;
+
+const NFTFactoryRequiredKeys = ['id', 'startingBlock', 'platformId', 'contractType', 'name', 'symbol', 'typeMetadata', 'standard', 'autoApprove', 'approved'];
+const NFTFactoryValidKeys = NFTFactoryRequiredKeys;
+
+const ArtistRequiredKeys = ['id'];
+const ArtistValidKeys = ['id', 'name'];
+
+const ProcessedTrackRequiredKeys = ['id'];
+const ProcessedTrackValidKeys = ['id', 'title', 'slug', 'description', 'websiteUrl', 'lossyAudioURL', 'lossyAudioIPFSHash', 'lossyArtworkURL', 'lossyArtworkIPFSHash'];
+
+type SeedPayload = {
+  entity: SeedEntity,
+  data: any
+}
+
+export const validateSeed = (payload: SeedPayload): void => {
+  entityValidator(payload);
+
+  const validatorFunctions = {
+    'platforms': [
+      () => minimumKeysPresent(payload, PlatformRequiredKeys),
+      () => onlyValidKeysPresent(payload, PlatformValidKeys),
+      () => typeValidator(payload, 'type', MusicPlatformType),
+    ],
+    'nftFactories': [
+      () => minimumKeysPresent(payload, NFTFactoryRequiredKeys),
+      () => onlyValidKeysPresent(payload, NFTFactoryValidKeys),
+      () => typeValidator(payload, 'contractType', NFTContractTypeName),
+      () => typeValidator(payload, 'standard', NFTStandard),
+    ],
+    'artists': [
+      () => minimumKeysPresent(payload, ArtistRequiredKeys),
+      () => onlyValidKeysPresent(payload, ArtistValidKeys),
+    ],
+    'processedTracks': [
+      () => minimumKeysPresent(payload, ProcessedTrackRequiredKeys),
+      () => onlyValidKeysPresent(payload, ProcessedTrackValidKeys),
+    ],
+  }
+
+  validatorFunctions[payload.entity].forEach((fn) => fn());
+}
+
+const minimumKeysPresent = (input: SeedPayload, keys: any): void => {
+  if (!keys.every((key: any) => input.data.hasOwnProperty(key))) {
+    throw new Error(`${input.entity} entity is missing required fields`)
+  }
+}
+
+const onlyValidKeysPresent = (input: SeedPayload, keys: any): void => {
+  if (!Object.keys(input.data).every((key: any) => keys.includes(key))) {
+    throw new Error(`${input.entity} entity has unsupported fields`)
+  }
+}
+
+const typeValidator = (input: SeedPayload, key: string, validOptions: any): void => {
+  if (!Object.values(validOptions).includes(input.data[key])) {
+    throw new Error(`not a valid ${input.entity} ${key}`)
+  }
+}
+
+const entityValidator = (input: SeedPayload): void => {
+  if (!Object.values(SeedEntities).includes(input.entity)) {
+    throw new Error('unknown seed entity');
+  }
+}
+
+export const persistSeed = async (payload: SeedPayload) => {
+  let message: any;
+  let dbClient: any;
+
+  entityValidator(payload);
+
+  if (['platforms', 'nftFactories'].includes(payload.entity)) {
```

Comment:
thinking on this further - matching this by type won't actually work, even for our short term needs in this PR. 

Consider:
 - in most existing migrations, we are inserting new platforms, so the getCrdtUpsertMessage is fine
 - in migration 31, we are updating existing platforms, so we do need getCrdtUpdateMessage on platforms

So we already need to support both to utilize this API for existing migrations.

I think we'll need to specify upsert vs update on the SeedPayload and support each and validate each separately (eg: for now, upsert on an nftFactory should require all keys, as it currently does, but update should only allow approved and autoApprove keys)